common/junit-platform-native/src/main/java/org/graalvm/junit/platform/JUnitPlatformFeature.java

```diff
@@ -156,20 +156,23 @@ private TestPlan registerTestPlan(Launcher launcher, List<Path> classpath) {
                 .map(Optional::get)
                 .filter(ClassSource.class::isInstance)
                 .map(ClassSource.class::cast)
-                .forEach((ClassSource classSource) -> {
-                    Class<?> clazz = classSource.getJavaClass();
-                    if (debug) {
-                        System.out.println("[Debug] Found test class: " + clazz);
-                    }
-                    RuntimeReflection.registerForReflectiveInstantiation(clazz);
-                    for (Field field : clazz.getDeclaredFields()) {
-                        RuntimeReflection.register(field);
-                    }
-                    for (Method method : clazz.getDeclaredMethods()) {
-                        RuntimeReflection.register(method);
-                    }
-                });
+                .forEach(this::registerTestClassForReflection);
 
         return testPlan;
     }
+
+    private void registerTestClassForReflection(ClassSource classSource) {
+        Class<?> clazz = classSource.getJavaClass();
+        if (debug) {
+            System.out.println("[Debug] Found test class: " + clazz);
+        }
```

Comment:
```suggestion
    private void registerTestClassForReflection(Class<?> clazz) {
        if (debug) {
            System.out.println("[Debug] Registering test class for reflection: " + clazz.getName());
        }
```

Reply:
Out of curiosity, why did you decide against `clazz.getName()`?

Reply:
Oh, snap! I was sure it was `clazz.getName()` before, my bad! Updating it now - it makes more sense to have `class.getName()`